### ISSUE TITLE
[Feature] list of reporting addresses to exclude from DMARC reporting

### ISSUE THREAD

**Issue by reinob** [rspamd/rspamd#5220]
## Summary

In addition (but similar) to the exclude_recipients option in DMARC reporting, there should be a way to have a map of "invalid" reporting addresses (RUA), as some always generate bounces.

## Motivation

In many cases the reporting address (e.g. "dmarc@mailingblue.com") is not directly associated with the recipient address, so that using exclude_recipients doesn't help.

## Describe alternatives you've considered

One could filter (DISCARD) e-mails to a list of addresses directly in e.g. postfix. But it would be easier, more maintainable, and more elegant to have that with rspamd.

I've looked in the lualib/rspamadm/dmarc_report.lua, but my LUA is too weak to provide such a patch.

**Comment by florianpiekert** [comment]
Looking for a solution to get rid of dmarc report bounces as well, I started setting up a map file for exclude_recipients.

From the manual on https://rspamd.com/doc/modules/dmarc.html:
"exclude_recipients (Optional): A path to a map file containing a list of **recipients to not send reports to**. Alternatively, an array of recipients can be used for the same purpose."

From your above report/comment I am confused whether in this map file the LOCAL recipients should be listed. I assumed the REMOTE dmarc_report target email addresses should be listed, so what is this additional configuration feature/directive you request?

**Comment by CacheMinimal** [comment]
I also wish to reduce the number of bounces/mailbox full messages I receive daily as postmaster from people who do not understand DMARC reporting!  I have configured rspamd's DMARC service as below (domain name changed for publication purposes):

`*** Section dmarc ***
reporting {
    enabled = true;
    email = 'postmaster@example.net';
    domain = 'example.net';
    org_name = 'Example Organisation';
    exclude_recipients = "/var/lib/rspamd/dmarc_exclude_recipients.map";
}

*** End of section dmarc ***`

The current content of my dmarc_exclude_recipients.map file is:

crm.delivrability@accor.com
dmarc-reports@alexandrasports.com
dmarc@alledm.cn
rua@dmarc.brevo.com
dmarc.ag@bt.com
dmarc@diysoup.com
dmarc-reports@gilluk.com
dmarc@mailinblue.com
dmarc@mcpolri.com
admin365@spinlock.co.uk
reports@tflcc.co.uk
reports@tflenforcement.com

This morning when the systemd timer ran to issue the 'rspamadm dmarc_report' command, I received NDRs from:

`[<dmarc@mailinblue.com>](mailto:dmarc@mailinblue.com): host aspmx.l.google.com[108.177.15.27] said: 550-5.7.1
    The user or domain that you are sending to (or from) has a policy that
    550-5.7.1 prohibited the mail that you sent. Please contact your domain
    550-5.7.1 administrator for further details. For more information, go to
    550 5.7.1  https://support.google.com/a/answer/172179

[reports@tflcc.co.uk](mailto:reports@tflcc.co.uk)
The email address you entered couldn't be found. Please check the recipient's email address and try to resend the message. If the problem continues, please contact your email admin.

[reports@tflenforcement.com](mailto:reports@tflenforcement.com)
The email address you entered couldn't be found. Please check the recipient's email address and try to resend the message. If the problem continues, please contact your email admin.

[dmarc.ag@bt.com](mailto:dmarc.ag@bt.com)
The recipient's mailbox is full and can't accept messages now. Please try resending your message later, or contact the recipient directly.`

Am I misunderstanding the use of exclude_recipients?  Is my map file somehow broken?  How can I further diagnose the fact that rspamd's DMARC reporting is still sending to these excluded recipients?

**Comment by fatalbanana** [comment]
Documentation for this option is apparently wrong (at least in part so; it's also described correctly).

`exclude_recipients` is applicable to *local* recipients for which DMARC reports shouldn't be generated (only the principal recipient is checked) - it is primarily useful for suppressing reporting from addresses which receive DMARC reports.

You can use `exclude_domains` to suppress reporting for *sender domains* with reporting addresses that bounce (or blackhole those addresses at the MTA).

**Comment by reinob** [comment]
@CacheMinimal,

The excluded_domains prevents DMARC reports to be sent when e-mails are received FROM one of the listed domains, i.e. you do not include "mailinblue.com" (unless you receive e-mails directly from them), but from the domain whose DMARC report lists a "bad" reporting address.

I use excluded_domains, and it works as advertised.

I would like to have something that works on DMARC-reporting-addresses (this is the feature request), i.e. where I can list "dmarc@mailingblue.com" so that no reports are (attempted to be) sent there, regardless of which sending domain happens to use that address as a reporting address. ← This is, as far as I can tell, not possible so far.

**Comment by fatalbanana** [comment]
> Taking my example from this morning, the NDR for `dmarc@mailinblue.com` was in response to a report generated for the domain `shows.bbc.co.uk`. To prevent this bounce in the future, do I add `mailinblue.com` or `shows.bbc.co.uk` to the exclude domains map?

The domain to list in `exclude_domains` should be `shows.bbc.co.uk` (or `bbc.co.uk`) in this instance.

**Comment by fatalbanana** [comment]
> I would like to have something that works on DMARC-reporting-addresses (this is the feature request), i.e. where I can list "[dmarc@mailingblue.com](mailto:dmarc@mailingblue.com)" so that no reports are (attempted to be) sent there, regardless of which sending domain happens to use that address as a reporting address. ← This is, as far as I can tell, not possible so far.

Yes, this is not readily possible- but it could be easily implemented. Short of having it you might blackhole these addresses at the MTA.
